content/blog/en/building-dojo-app.md

```diff
@@ -0,0 +1,1124 @@
+		break;
+	case 'large':
+		icache.set(
+			'currentPrice',
+			price + 1
+		);
+		break;
+}
+```
+
+Finally, once the user is done customizing their Drink, they need a way to add that drink to the Cart.
+
+```tsx
+// src/widgets/DrinkPage.tsx
+
+...
+
+export default factory(function DrinkPage({
+	middleware: { icache, injector, store },
+	properties,
+}) {
+	...
+
+	return (
+		<div classes={[css.root]}>
+			...
+			<Button
+				onClick={() => {
+					const size = icache.getOrSet('size', 'small');
+					const addins = icache.getOrSet('addins', []);
+					const toppings = icache.getOrSet('toppings', []);
+					const flavors = icache.getOrSet('flavors', []);
+					const currentPrice = icache.getOrSet('currentPrice', price);
+
+					const userDrink: Drink = {
+						...drink,
+						price: currentPrice,
+						size,
+						addins,
+						toppings,
+						flavors,
+					};
+					add(userDrink);
+					if (router) {
+						router.setPath('menu');
+					}
+				}}
+			>
+				Add to Cart
+			</Button>
+		</div>
+	);
+});
+```
+
+Here, we have a `Button` that when clicked is going to create an updated `Drink` object based on the current selections, add that drink to the Cart, and the navigate to the `menu` route.
+
+```ts
+add(userDrink);
+if (router) {
+	router.setPath('menu');
+}
+```
+
+## Show me the Cart
+
+The `Cart` widget is a basic list of the Drinks that are in the Cart. It provides a look at the total price for all the drinks and a way to remove drinks from the Cart.
+
+```tsx
+// src/widgets/Cart.tsx
+import { create, tsx } from '@dojo/framework/core/vdom';
+import store from '../middleware/cartStore';
+import { initCart, removeFromCart } from '../processes/cart';
+
+import * as css from './styles/Cart.m.css';
+
+const factory = create({ store });
+
+export default factory(function Cart({ middleware: { store } }) {
+	const { executor, get, path } = store;
+	const init = executor(initCart);
+	const remove = executor(removeFromCart);
+	const orders = get(path('drinks'));
+	const total = get(path('total')) || 0;
+	if (!orders) {
+		init({});
+	}
+
+	return (
+		<div classes={[css.root]}>
+			{!orders || !orders.length ? (
+				<div>Cart is empty</div>
+			) : (
+				<div key="drink-orders" classes={[css.content]}>
+					{orders.map((x) => (
+						<div classes={[css.order]} key={`order-${x.id}`}>
+							<div classes={[css.item]}>
+								<label>{x.name}</label>
+								<label>Size: {x.size}</label>
+								<button
+									classes={[css.remove]}
+									onclick={() => {
+										remove(x);
+									}}
+								>
+									Remove
+								</button>
+							</div>
+						</div>
+					))}
+				</div>
+			)}
+			<label>Total: ${total.toFixed(2)}</label>
+		</div>
+	);
+});
+```
+
+It's not too fancy, but it's only job is to display the Drinks in the Cart and let the user remove Drinks from the Cart. In this application, the Cart is not its own page, it will be displayed in a `SlidePane` so that it's visible on any page the user is on.
```

Comment:
```suggestion
It's not too fancy, but its only job is to display the Drinks in the Cart and let the user remove Drinks from the Cart. In this application, the Cart is not its own page, it will be displayed in a `SlidePane` so that it's visible on any page.
```